content/2018/opening-other-file-formats.md

```diff
@@ -44,30 +44,38 @@ You can find information for writing your own `ftplugin` files at `:h write-file
     endif
     let b:did_ftplugin = 1
 
-to prevent the `ftplugin` from running multiple times on the same buffer. You can then specify commands to be executed on the filetype in question when detected.
+to prevent the `ftplugin` from running multiple times on the same buffer. You can then specify commands to be executed on the filetype in question when detected. If you find this a bit cumbersome, you're not alone. You can bypass the need for such a guard by instead using the `~/.vim/after/ftplugin` directory, which executes after the default `ftplugin` runs (if there is one).
 
 ## Putting it all together
 
 Coming back to our opening statements, how do we make this work for external formats? For starters, make your `ftdetect` file
 
-    :!mkdir -p ~/.vim/ftdetect/pdf.vim
+    :!mkdir -p ~/.vim/ftdetect/video.vim
 
 and edit it with extension detection parameters
 
-    au BufRead,BufNewFile *.pdf set filetype=pdf
+    au BufRead,BufNewFile *.avi,*.mp4,*.mkv,*.mov,*.mpg set filetype=video
 
 then create your `ftplugin` file
 
-    :!mkdir -p ~/.vim/ftplugin/pdf.vim
+    :!mkdir -p ~/.vim/after/ftplugin/video.vim
 
 and edit it appropriately:
 
-    if exists("b:did_ftplugin")
-      finish
-    endif
-    let b:did_ftplugin = 1
-    sil exe "!zathura " . shellescape(expand("%:p")) . " &" | b# | bd# | redraw! | syn on
+    sil exe "!mplayer " . shellescape(expand("%:p")) . " &>/dev/null &" | b# | bd# | redraw! | syn on
+
+In this example we're using the PDF reader `mplayer` to open the PDF file, but you could just as easily use any PDF viewer. This silently executes an `!external-program` on the current buffer and places it in the background (`&`), discarding any output or errors to `/dev/null`. The `|`s could easily be separated onto multiple lines and act as a list of sequential commands to switch back to the previous buffer (`b#`), delete the PDF buffer (`bd#`), and then ensure that the screen was not garbled in the process by redrawing and ensuring syntax highlighting is on.
```

Comment:
You still have "PDF reader" and "PDF", here.

Reply:
Woops!